compiler/src/iree/compiler/ExternalInterfaces/UtilExternalModels.cpp

```diff
@@ -366,8 +366,8 @@ void registerUtilExternalModels(DialectRegistry &registry) {
         // Never hoist empty and other pure metadata ops as a leaf. It's fine to
         // hoist them as a part of a larger constant tree that does actual work.
         HoistableNonLeafOpInterfaceHelper<
-            tensor::EmptyOp, tensor::ExpandShapeOp,
-            tensor::CollapseShapeOp>::registerOpInterface(context);
+            tensor::EmptyOp, tensor::ExpandShapeOp, tensor::CollapseShapeOp,
+            tensor::ExtractSliceOp>::registerOpInterface(context);
```

Comment:
Minor comment, I would have made this only not hoist if the source of the slice has multiple uses. Probably not a common case so not a big deal.

Reply:
Is there a case you are thinking of where it would be beneficial to hoist `extract_slice` (with 1 use)? The only advantage I could think of would be to reduce the # of ops in non-initializers. But I don't see how this would be any different from hoisting collapse shape, expand shapes, etc.


I can definitely change this, it doesn't seem like a big change.

Reply:
Mainly just that taking a slice yields a smaller tensor (unless it's rank reducing and dropping unit dims, in which case it's basically a reshape), and I would assume smaller contiguous tensor is better than the larger thing. I also meant the source of the `extract_slice` having 1 use, not the slice itself, which is probably not a common case so not high priority.